linera-sdk-derive/src/utils.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use convert_case::{Case, Casing};
+use proc_macro2::{Ident, Span};
+
+/// Extracts the first `Ident` in a type and convert to snake case.
+pub fn snakify(ident: &Ident) -> Ident {
```

Comment:
Looks like we should choose `convert_case` and [`heck`](https://docs.rs/heck/latest/heck/) also used in this repo.

Reply:
I'm not sure I understand what you mean?

Reply:
Oops. I fixed the comment